File: plugins/flutter_bg_location_plugin/ios/Classes/Services/LocationStorage.swift
import Foundation

/// iOS-эквивалент Android LocationStorage на базе UserDefaults
class LocationStorage {
    static let shared = LocationStorage()
    private let defaults = UserDefaults.standard
    private init() {}

    // MARK: - Keys и значения по умолчанию
    private enum Keys {
        static let hash = "hash"
        static let tickerSeconds = "ticker_seconds"
        static let tickersLeft = "tickers_left"
        static let tickerId = "ticker_id"
        static let orderId = "order_id"
    }
    private let defaultTickerSeconds = 10

    // MARK: - Хэш
    func getHash() -> String {
        return defaults.string(forKey: Keys.hash) ?? ""
    }

    func setHash(_ hash: String) {
        defaults.set(hash, forKey: Keys.hash)
    }

    // MARK: - Секунды для тикера
    func getTickerSeconds() -> Int {
        if let value = defaults.object(forKey: Keys.tickerSeconds) as? Int {
            return value
        }
        return defaultTickerSeconds
    }

    func setTickerSeconds(_ seconds: Int) {
        defaults.set(seconds, forKey: Keys.tickerSeconds)
    }

    // MARK: - Количество тикеров
    func getTickers() -> Int {
        return defaults.integer(forKey: Keys.tickersLeft)
    }

    func setTickers(_ count: Int) {
        defaults.set(count, forKey: Keys.tickersLeft)
    }

    func declineOneTickers() {
        let current = getTickers()
        defaults.set(max(current - 1, 0), forKey: Keys.tickersLeft)
    }

    // MARK: - ID тикера
    func getTickerId() -> String {
        return defaults.string(forKey: Keys.tickerId) ?? ""
    }

    func setTickerId(_ id: String) {
        defaults.set(id, forKey: Keys.tickerId)
    }

    // MARK: - ID ордера
    func getOrderId() -> Int {
        return defaults.integer(forKey: Keys.orderId) ?? 0
    }

    func setOrderId(_ id: Int) {
        defaults.set(id, forKey: Keys.orderId)
    }
}

File: plugins/flutter_bg_location_plugin/android/src/main/kotlin/com/ezride/flutter_bg_location_plugin/services/LocationStorage.kt
package com.ezride.flutter_bg_location_plugin.services

import android.content.Context
import android.content.SharedPreferences

class LocationStorage(context: Context) {
    companion object {
        private const val PREFS_NAME = "bg_loc_prefs"                   // идентификатор преференса
        private const val KEY_HASH = "hash"                             // ключ хэша по которому будет отправляться данные на сервер
        private const val KEY_TICKER_SECONDS = "ticker_seconds"         // ключ секунд для 1 тика 
        private const val DEFAULT_TICKER_SECONDS = 10                   // дефолтное значение секунд для 1 тика 
        private const val KEY_TICKERS_LEFT = "tickers_left"             // ключ сколько осталось тикеров
        private const val KEY_TICKER_ID = "ticker_id"                   // ключ id тикера к чему он подвязан если активен
        private const val KEY_ORDER_ID = "order_id"                     // ключ id ордера к чему он подвязан если активен
        
    }
    private val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)


    //////////////Хэш
    // запрос хэша
    fun getHash(): String? = prefs.getString(KEY_HASH, "");
    //сохранение хэша
    fun setHash(hash: String) {
        prefs.edit().putString(KEY_HASH, hash).apply();
    }

    //////////////Тикер секунд
    // запрос секунд на 1 тикер 
    fun getTickerSeconds(): Int = prefs.getInt(KEY_TICKER_SECONDS, DEFAULT_TICKER_SECONDS);
    //сохранение тикера секунд
    fun setTickerSeconds(tickerSeconds: Int) {
        prefs.edit().putInt(KEY_TICKER_SECONDS, tickerSeconds).apply()
    }

    //////////////Колличественное свойство тикеров
    // запрос тикеров 
    fun getTickers(): Int = prefs.getInt(KEY_TICKERS_LEFT, 0);
    //сохранение тикеров
    fun setTickers(tickerCount: Int) {
        prefs.edit().putInt(KEY_TICKERS_LEFT, tickerCount).apply()
    }
    //уменьшение тикера на 1
    fun declineOneTickers() {
        val tickers = getTickers();
        prefs.edit().putInt(KEY_TICKERS_LEFT, tickers-1).apply()
    }

    //////////////Подвязка тикера локации к данным
    // запрос айди 
    fun getTickerId(): String? = prefs.getString(KEY_TICKER_ID, "");
    //сохранение айди тикера
    fun setTickerId(tickerId : String) {
        prefs.edit().putString(KEY_TICKER_ID, tickerId).apply()
    }

    //////////////Подвязка тикера локации к данным
    // запрос айди 
    fun getOrderId(): Int = prefs.getInt(KEY_ORDER_ID, 0);
    //сохранение айди тикера
    fun setOrderId(orderId : Int) {
        prefs.edit().putInt(KEY_ORDER_ID, orderId).apply()
    }
}